### ISSUE TITLE
[java] Make UnusedFormalParameter consider enum classes

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3628]
**Affects PMD Version:6.39.0** 

**Rule:UnusedFormalParameter**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedformalparameter>

**Description:**
Hi! I think PMD should report a warning at line 5 because there is a unused formal parameter `arg`, but it does not. Hence, I think this is a false negative. Thanks for your consideration. I am willing to help fix this issue.

**Code Sample demonstrating the issue:**

```java
class Tester {
  enum EnumClass {
    FOO;
    EnumClass() {}
    private void func(int arg) {}  // should report a warning here
  }
}
```

**Expected outcome: A warning**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by adangel** [comment]
Yes, that's a false negative. I confirmed, that it is *not* fixed with #3619. So a PR to fix this issue would be appreciated.

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
